qiskit/providers/ibmq/api_v2/clients/websocket.py

```diff
@@ -150,64 +153,81 @@ def get_job_status(self, job_id, timeout=None):
 
-        original_timeout = timeout
-        start_time = time.time()
-        last_status = None
-
-        try:
-            # Read messages from the server until the connection is closed or
-            # a timeout has been reached.
-            while True:
-                try:
-                    with warnings.catch_warnings():
-                        # Suppress websockets deprecation warnings until the fix is available
-                        warnings.filterwarnings("ignore", category=DeprecationWarning)
-                        if timeout:
-                            response_raw = yield from asyncio.wait_for(
-                                websocket.recv(), timeout=timeout)
-
-                            # Decrease the timeout, with a 5-second grace period.
-                            elapsed_time = time.time() - start_time
-                            timeout = max(5, int(original_timeout - elapsed_time))
-                        else:
-                            response_raw = yield from websocket.recv()
-                    logger.debug('Received message from websocket: %s',
-                                 response_raw)
-
-                    response = WebsocketMessage.from_bytes(response_raw)
-                    last_status = response.data
-
-                    job_status = response.data.get('status')
-                    if (job_status and
-                            ApiJobStatus(job_status) in API_JOB_FINAL_STATES):
-                        break
-
-                except futures.TimeoutError:
-                    # Timeout during our wait.
-                    raise WebsocketTimeoutError('Timeout reached') from None
-                except ConnectionClosed as ex:
-                    # From the API:
-                    # 4001: closed due to an internal errors
-                    # 4002: closed on purpose (no more updates to send)
-                    # 4003: closed due to job not found.
-                    message = 'Unexpected error'
-                    if ex.code == 4001:
-                        message = 'Internal server error'
-                    elif ex.code == 4002:
-                        break
-                    elif ex.code == 4003:
-                        message = 'Job id not found'
-                    raise WebsocketError('Connection with websocket closed '
-                                         'unexpectedly: {}'.format(message)) from ex
-        finally:
-            with warnings.catch_warnings():
-                # Suppress websockets deprecation warnings until the fix is available
-                warnings.filterwarnings("ignore", category=DeprecationWarning)
-                yield from websocket.close()
-
-        return last_status
+        current_retry = 0
+        while current_retry <= retry_max:
+            websocket = yield from self._connect(url)
+
+            connection_closed_ex_code = None
+            original_timeout = timeout
```

Comment:
You're essentially doubling the user specified timeout value by resetting it every time

Reply:
Didn't actually catch that, thanks Jessie!